crates/witness-generator/src/eest_generator.rs

```diff
@@ -385,4 +381,40 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn test_invalid_block() -> Result<()> {
```

Comment:
An extra test to dig more into the details on the partial witness from an invalid block. For this case, we should see the witness to be filled but only with the parent header data since the block is invalid due to parent block checks. The rest of the witness fields must be empty.